compiler/src/config/mod.rs

```diff
@@ -0,0 +1,113 @@
+pub(crate) mod common;
+pub(crate) mod compiler;
+
+use std::path::{Path, PathBuf};
+use serde::{Serialize, Deserialize};
+use figment::{Figment, providers::Yaml, providers::Format};
+use std::fs;
+use serde_yaml;
+
+use compiler::Compiler;
+use common::Common;
+
+
+#[derive(Debug, Serialize, Deserialize)]
+pub (crate) struct Config {
+    pub common: Common,
+    pub gcc: Compiler,
+    pub rustc: Compiler,
+}
+
+impl Config {
+    pub fn new(common: Common,
+               gcc: Compiler,
+               rustc: Compiler) -> Self {
+        Self {
+            common,
+            gcc,
+            rustc
+        }
+    }
+}
+
+
+pub(crate) fn load_config(config_file_path: PathBuf) -> Result<Config, &'static str> {
+    
+    if !Path::new(config_file_path.to_str().unwrap()).exists() {
+        let config = load_default_config().unwrap(); 
+
+        write_config(&config, &config_file_path).unwrap();
+
+        return Ok(config)       
+    }
+    
+    match Figment::new().merge(Yaml::file(config_file_path.clone())).extract() {
+        Ok(config) => {
+            return Ok(config)
+        }
+
+        Err(_) => {
+            let config = load_default_config().unwrap(); 
+
+            write_config(&config, &config_file_path).unwrap();
+
+            return Ok(config) 
+        }
+    }
+    
+    /*
+    let config = Figment::new()
+                    .merge(Yaml::file(config_file_path))
+                    .extract().unwrap(); 
+    */
+
+        
+}
+
+
+pub(crate) fn load_default_config() -> Result<Config, &'static str> {
```

Comment:
not sure if this function is loading anything, probably `get_default_config` can me more descriptive

Reply:
Completely agree with you. The idea was to load default config from the some kind of "secret" storage (binary, maybe). But now, you are right - config is not "loaded".

Reply:
Rename to get_default_config